src/platform_impl/linux/wayland/window.rs

```diff
@@ -59,15 +59,25 @@ impl Window {
         // Create the surface first to get initial DPI
         let window_store = evlp.store.clone();
         let cursor_manager = evlp.cursor_manager.clone();
-        let surface = evlp.env.create_surface(move |dpi, surface| {
-            window_store.lock().unwrap().dpi_change(&surface, dpi);
-            surface.set_buffer_scale(dpi);
+        let surface = evlp.env.create_surface(move |scale_factor, surface| {
+            window_store
+                .lock()
+                .unwrap()
+                .scale_factor_change(&surface, scale_factor);
+            surface.set_buffer_scale(scale_factor);
         });
 
-        let dpi = get_dpi_factor(&surface) as f64;
+        let scale_factor = {
+            if let Some(output) = get_outputs(&surface).last() {
```

Comment:
This is incorrect, initial scale factor on Wayland is `1`and we should trust compositor events, don't guess it by monitor, this won't work. Right now I create a wrong window size on startup, which is larger then my output, because I'm drawing 8k window on 4k screen with your PR. For more https://gitlab.freedesktop.org/wayland/wayland/issues/133

Reply:
Okay, not clear if/how there is a way to avoid drawing a buffer with the wrong scale.
The get_dpi_factor is misleading since it use surface enter to compute dpi.
That's why I thought we are trying to get a better guess but failing.
Changed the init to make it clear we use an initial 1x scale factor.

Reply:
>Okay, not clear if/how there is a way to avoid drawing a buffer with the wrong scale.

I've posted you a link, which is telling that it's impossible due to protocols.

>Changed the init to make it clear we use an initial 1x scale factor.

Please, return to old logic, since it was a correct logic and more clear, just hard coding `1` isn't an option. You can add a comment explaining that most of the time it'll be `1` due to how protocols are designed now, and that we get proper scale factor after `enter`.

Reply:
I feel like if there is a single output we can't go wrong using it.
Reverted to scale_factor=1 constant variable as requested.